internal/pkg/opentelemetry/controller.go

```diff
@@ -30,21 +30,35 @@ type Controller struct {
 	logger         logr.Logger
 	version        string
 	tracerProvider trace.TracerProvider
-	tracersMap     map[string]trace.Tracer
+	tracersMap     map[tracerID]trace.Tracer
 	bootTime       int64
 }
 
-func (c *Controller) getTracer(pkg string) trace.Tracer {
-	t, exists := c.tracersMap[pkg]
+type tracerID struct{ name, version, schema string }
+
+func (c *Controller) getTracer(pkg, tracerName, version, schema string) trace.Tracer {
+	tID := tracerID{name: pkg, version: c.version}
+	if tracerName != "" {
+		tID = tracerID{name: tracerName, version: version, schema: schema}
+	}
+
+	t, exists := c.tracersMap[tID]
 	if exists {
 		return t
 	}
 
-	newTracer := c.tracerProvider.Tracer(
-		"go.opentelemetry.io/auto/"+pkg,
-		trace.WithInstrumentationVersion(c.version),
-	)
-	c.tracersMap[pkg] = newTracer
+	var newTracer trace.Tracer
+	if tracerName != "" {
+		// If the user has provided a tracer, use it.
+		newTracer = c.tracerProvider.Tracer(tracerName, trace.WithInstrumentationVersion(version), trace.WithSchemaURL(schema))
+	} else {
+		newTracer = c.tracerProvider.Tracer(
+			"go.opentelemetry.io/auto/"+pkg,
+			trace.WithInstrumentationVersion(c.version),
+		)
```

Comment:
```suggestion
			trace.WithInstrumentationVersion(c.version),
			trace.WithSchemaURL(schema))
		)
```

Reply:
Relates to the above comment. `schema` here is the user-provided schema (if exists). Hence, if we provide a schema here it should be something related to our distro (since this case deals with all the fully auto-generated spans).

Reply:
That makes sense. Can we then set `semconv.SchemaURL` from [`semconv/v1.26.0`](https://pkg.go.dev/go.opentelemetry.io/otel/semconv/v1.26.0)?

Reply:
We set it here: https://github.com/open-telemetry/opentelemetry-go-instrumentation/blob/c6620c7f9f80bcd1ad6fc59278a469d0a743da15/instrumentation.go#L285

I'm not sure about the differences between schema URL for the tracerProvider and the tracer, and in our case will they be the same or not?

Reply:
That schema URL is for resource which indicates what semantic conventions the resource itself contain: https://github.com/open-telemetry/semantic-conventions/blob/main/docs/resource/README.md

The `Tracer` schema URL indicates what semantic convention version the span data is recorded with. This is the field that populates this in OTLP: https://github.com/open-telemetry/opentelemetry-proto-go/blob/109641596aab69d0bec6e1f2856ecf749de88b67/otlp/trace/v1/trace.pb.go#L386-L390

Reply:
I see.
Setting the schema URL here from the latest semconv assumes all the probes are updated to the latest semconv right? This is currently the case but will probably need some maintenance, and each probe can have a different version as well.

Reply:
> This is currently the case but will probably need some maintenance, and each probe can have a different version as well.

True. We will need to be diligent about this and likely need to build some testing to ensure this is accurate.